common/txmgr/confirmer.go

```diff
@@ -435,6 +448,55 @@ func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) Che
 	return nil
 }
 
+func (ec *Confirmer[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, R, SEQ, FEE]) ProcessStuckTransactions(ctx context.Context, blockNum int64) error {
+	// Use the detector to find a stuck tx for each enabled address
+	stuckTxs, err := ec.stuckTxDetector.DetectStuckTransactions(ctx, ec.enabledAddresses, blockNum)
+	if err != nil {
+		return fmt.Errorf("failed to detect stuck transactions: %w", err)
+	}
+	if len(stuckTxs) == 0 {
+		return nil
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(len(stuckTxs))
+	errorList := []error{}
+	var errMu sync.Mutex
+	for _, tx := range stuckTxs {
+		// All stuck transactions will have unique from addresses. It is safe to process separate keys concurrently
+		// NOTE: This design will block one key if another takes a really long time to execute
+		go func(tx txmgrtypes.Tx[CHAIN_ID, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) {
+			defer wg.Done()
+			lggr := tx.GetLogger(ec.lggr)
+			// Create an purge attempt for tx
+			purgeAttempt, err := ec.TxAttemptBuilder.NewPurgeTxAttempt(ctx, tx, lggr)
```

Comment:
What happens if gas bumping fails due to reaching the max limit?

Reply:
I think we could consider this as a misconfiguration since our attempt threshold config would have to be higher than the amount of bumped attempts it takes to hit the configured max price. I don't think we can really get around this in code since our hands are tied with configs. I can add a note to the `AutoPurge.MinAttempts` config to call this out though

Reply:
I feel like ignoring configs could look like unexpected behavior and maybe cause confusion. Since NOPs wouldn't know the inner workings of this auto-purge feature, I could foresee them noticing a tx post with a higher price than their configs and falsely raise a flag only for us to debug and find it was expected.